www/viewgame

```diff
@@ -1178,10 +1178,17 @@ if (count($links) == 0 && $dlnotes == "") {
         echo "</td><td><div class='downloaditem' id='dlItem$i'>
               <a href=\"$url\"><b>$linktitle</b></a>$zipstar<br>";
 
-        if ($compPri)
+        $unbox_supported = [17, 22, 23, 41];
+
+        if (in_array($fmtcomp, $unbox_supported) && (strpos($url, 'https://ifarchive.org') === 0)) {
```

Comment:
IF Archive links in the database aren't normalized. I see `http://` and `www`, so this prefix test doesn't currently catch everything.

I suggest normalizing them when saving, and updating the database separately.

Reply:
We do normalize them! At line 1155, we run `urlToMirror`, which used to allow users to select an IF Archive URL, but now it just runs this regex:

```
$result = preg_replace('!^https?://((www\.|mirror\.)?ifarchive(\.[a-z]+?)?\.org)/!', 'https://ifarchive.org/', $url);
```

Reply:
Nice! BTW, that regex is repeated in what appears to be a similar unboxing operation in line 933:
```php
$unboxUrl = false;
if ($link['compression'] && $link['compressedprimary']) {
    $open = urlencode($link['compressedprimary']);
    if (preg_match('!^https?://(www\.|mirror\.)?ifarchive\.org/!', $link['url'])) {
        $linkUrl = urlencode($link['url']);
        $unboxUrl = "https://unbox.ifarchive.org/?url=$linkUrl&open=$open";
```